### ISSUE TITLE
pihole update broken - FTLDNS

### ISSUE THREAD

**Issue by PromoFaux** [pi-hole/pi-hole#2068]
@EngieDev commented on [Thu Mar 29 2018](https://github.com/pi-hole/FTL/issues/245)

**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*
- [x] The issue I am reporting isn't a duplicate 

**How familiar are you with the codebase?:**
1
---

**[BUG | ISSUE] Expected Behaviour:**
`pi-hole -up` should update correctly 

**[BUG | ISSUE] Actual Behaviour:**
Fails to update (or to indicate that its updated)
DNS error since FTL gets started after the update checks, leaving no DNS for the web interface update.

**[BUG | ISSUE] Steps to reproduce:**
Run `pihole -up` several times and FTL will always say "update available"

Fresh raspbian stretch install, with pihole installed first, then the core/web FTLDNS was activated with:
```
echo "FTLDNS" | sudo tee /etc/pihole/ftlbranch
pihole checkout core FTLDNS 
pihole checkout web FTLDNS
```

**Log file output [if available]**
```pi@raspberrypi:~ $ pihole -up
  [i] Checking for updates...
  [i] Pi-hole Core:     up to date
  [i] FTL:              update available

  [i] FTL out of date

  [i] FTL Checks...
  [✓] Detected ARM-hf architecture (armv6 or lower) Using ARM binary
  [i] Checking for existing FTL binary...
  [✓] Downloading and Installing FTL


fatal: unable to access 'https://github.com/pi-hole/AdminLTE.git/': Could not resolve host: github.com
  [i] Web Interface:    up to date

  [i] FTL version is now at
  [✓] Starting pihole-FTL service
  [✓] Enabling pihole-FTL service to start on reboot
```

**Device specifics**

Hardware Type: Raspberry Pi Generation 1 Model B - Arm6 CPU
OS: Raspbian Stretch

---

@PromoFaux commented on [Fri Mar 30 2018](https://github.com/pi-hole/FTL/issues/245#issuecomment-377520663)

The update function isn't really made to work with alternative branches, as it works on the tagged releases, something we have not done for the `FTLDNS` branches. to update, you'll need to manually head into `/etc/.pihole` and run `sudo git pull`, followed by a `pihole -r` to install the latest.

---

@DL6ER commented on [Fri Mar 30 2018](https://github.com/pi-hole/FTL/issues/245#issuecomment-377639192)

@PromoFaux The issue seems to be that we disable FTL for updating it and only re-enable it at the end of the updater - leaving the majority of the update script without resolving power. This does not seem to be limited to special branches and needs some work on. However, this is a core bug.

---

@PromoFaux commented on [Sat Mar 31 2018](https://github.com/pi-hole/FTL/issues/245#issuecomment-377685951)

Ah, OK. Moving to core...

**Comment by technicalpyro** [comment]
odd I managed to run `pihole -up` successfully yesterday

**Comment by EngieDev** [comment]
Just a note, the same thing appears to occur when you run `pihole -r`. The FTL binary gets reinstalled, however gets started at the end, leaving no resolving power for gravity.

**Comment by neauoire** [comment]
Just tried `pihole -up` and received the `Unable to complete update, please contact Pi-hole Support`.

**Comment by PromoFaux** [comment]
Yeah, `pihole -up` in it's current state wont work with the `FTLDNS` branch, however, it will do once [this pr](https://github.com/pi-hole/pi-hole/pull/2077) is merged.

**Comment by SanderGit** [comment]
This just merged and I can confirm it fixed the problem.

**Comment by Mcat12** [comment]
Should be fixed by #2077
